docs/static/troubleshooting.asciidoc

```diff
@@ -0,0 +1,270 @@
+[[troubleshooting]] 
+== Troubleshooting Common Problems
+
+We will be adding more tips and solutions, so please check back soon.
+
+Also check out the https://discuss.elastic.co/c/logstash[Logstash discussion
+forum].
+
+
+[float] 
+[[ts-install]] 
+== Installation and setup
+
+
+[float] 
+[[ts-temp-dir]] 
+=== Inaccessible temp directory
+
+Certain versions of the JRuby runtime and libraries
+in certain plugins (the Netty network library in the TCP input, for example) copy
+executable files to the temp directory. This situation causes subsequent failures when
+`/tmp` is mounted `noexec`. 
+
+*Possible solutions*
+
+* Change setting to mount `/tmp` with `exec`.
+* Specify an alternate directory using the `-Djava.io.tmpdir` setting in the `jvm.options` file.
+ 
+
+[float] 
+[[ts-ingest]] 
+== Data ingestion
+
+[float] 
+[[ts-429]] 
+=== Error response code 429
+
+A `429` message indicates that an application is busy handling other requests. For
+example, Elasticsearch throws a `429` code to notify Logstash (or other indexers)
+that the bulk failed because the ingest queue is full. Any documents that
+weren't processed should be retried.
+
+TBD:  Does Logstash retry? Should the user take any action?
```

Comment:
logstash will retry forever here. potential actions would be to check how es is behaving, we can link to the elasticsearch monitoring docs https://www.elastic.co/guide/en/elasticsearch/reference/current/es-monitoring.html
and also maybe some non-xpack apis too: https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-stats.html